src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -114,6 +104,51 @@ public synchronized static void shutdownOpenShiftClient() {
     }
   }
 
+  /*Right now, when the buildconfig name is different then git repo name
+  then sync plugin is not able to find buildconfig by repo name. While
+  syncing back to openshift after creating a buildconfig with strategy
+  JenkinsPipeline, and create a new buildconfig repo name.
+  Also at the time of master commit, it is not able to find the buildConfig
+  to start a new build and failing. To fix this
+  issue, added a label of gitRepository name so it will first try to
+  find the BuildConfig with name of jenkins job, if that is not present
+  that is our case then it will also try to find the buildconfig having
+  label with the name of jenkins job
+  if found return the buildConfig after both checks otherwise null
+  */
+  public static BuildConfig getBuildConfig(String namespace, String buildConfigname) {
+
+    logger.info("Finding BuildConfig for namespace: " + namespace + " name: " + buildConfigname);
+
+    BuildConfig jobBuildConfig = getOpenShiftClient().buildConfigs().
+      inNamespace(namespace).withName(buildConfigname).get();
+
+    if (jobBuildConfig == null){
+
+      logger.info("Not able to find BuildConfig for namespace: " + namespace + " name: " + buildConfigname);
+
+      logger.info("Finding BuildConfig for namespace: " + namespace + " with label gitRepository: " + buildConfigname);
+
+      /* BuildConfig and Repo Name are not same, lets find the buildConfig by label*/
+      BuildConfigList jobBuildConfigs = getOpenShiftClient().buildConfigs().inNamespace(namespace)
```

Comment:
I need to fix this in our repo as well, as I forgot to handle this when I picked the original commit from you all that introduced this, but we should be using https://github.com/openshift/jenkins-sync-plugin/blob/master/src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java#L154 here 

Even if you all don't want to port over the `CredentialUtils.getCurrentToken()` piece, a place holder/skeleton version of `getAuthenticatedOpenShiftClient`  would be good so that when we pull such commits into upstream, we can pick these changes as-is